velox/vector/tests/VectorTest.cpp

```diff
@@ -3899,5 +3899,21 @@ TEST_F(VectorTest, testOverSizedArray) {
   EXPECT_THROW(BaseVector::flattenVector(constArray), VeloxUserError);
 }
 
+TEST_F(VectorTest, testSliceBuffer) {
+  auto bufferPtr = AlignedBuffer::allocate<int64_t>(10, pool());
+  auto sliceBufferPtr =
+      BaseVector::sliceBuffer(*BIGINT(), bufferPtr, 1, 5, pool());
+  ASSERT_TRUE(sliceBufferPtr->isView());
+  ASSERT_EQ(sliceBufferPtr->as<int64_t>(), bufferPtr->as<int64_t>() + 1);
+
+  bufferPtr = AlignedBuffer::allocate<bool>(16, pool());
+  sliceBufferPtr = BaseVector::sliceBuffer(*BOOLEAN(), bufferPtr, 8, 8, pool());
+  ASSERT_TRUE(sliceBufferPtr->isView());
+  ASSERT_EQ(sliceBufferPtr->as<bool>(), bufferPtr->as<bool>() + 1);
+
+  sliceBufferPtr = BaseVector::sliceBuffer(*BOOLEAN(), bufferPtr, 5, 5, pool());
```

Comment:
What if offset / length are out of range?

Reply:
The original logic did not have any check. I added checks to ensure the parameters are within the correct range and also added tests to verify it.